### PR TITLE
#1387: fix error after switching on saved view

### LINKED ISSUE #1387: "Uncaught TypeError: Cannot read property 'complete' of undefined" appears in dev console if save Previewed image as a new View and open this View on another page
### Preconditions (*)
1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset

### Steps to reproduce (*)
1. Go to **Content - Media Gallery**
2. Click **Search Adobe Stock**
3. Go to the first page from the grid, and select any image to open its Preview
4. Click on **Default View** (bookmarks view)
5. Click **Save View As...**, 
6. Click **Save all Changes** :arrow_right: 
7. Switch to the _Default View_
8. Go to the second page in Adobe Stock grid
9. Switch to the previously created _New View_

### Expected result (*)
No error message in dev console

### Actual result (*)
:x: _Uncaught TypeError: Cannot read property 'complete' of undefined_ appears in dev console 
uncought_complete_view_2520

### PR DESCRIPTION
**Pull request by mmezhensky**
# Description (*)

This PR fixed issue [1387](https://github.com/magento/adobe-stock-integration/issues/1387).

### Fixed Issues

1. Fixes magento/adobe-stock-integration#1387: "Uncaught TypeError: Cannot read property 'complete' of undefined" appears in dev console if save Previewed image as a new View and open this View on another page

### Manual testing scenarios (*)

1. Go to Content - Media Gallery
2. Click Search Adobe Stock
3. Go to the first page from the grid, and select any image to open its Preview
4. Click on Default View (bookmarks view)
5. Click Save View As...,
6. Click Save all Changes arrow_right
7. Switch to the Default View
8. Go to the second page in Adobe Stock grid
9. Switch to the previously created New View

**>> No error message in dev console**

### CODE REVIEW

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @mmezhensky, thanks for the pull request!

I've updated the issue with a comment, but basically we have another bug when performing the same steps described on the issue. The bug is the image that is displayed after switching bookmarks is the wrong one. Follow below some suggestions on these issues:

So, related to the console error, I believe we should handle that on magento2 repository, and on the `app/code/Magento/Ui/view/base/web/js/grid/columns/image-preview.js` component, to solve the issue we can extract the part that is causing the error to an external function and handle it there. The problem seems to be related to the image not being there when the code gets executed, we could do something like this to prevent the error from happening:

```
        updateUserInterface: function () {
            var img = $(this.previewImageSelector + ' img');

            if (!img.get(0)) {
                setTimeout(function () {
                    this.updateUserInterface();
                }.bind(this), 100);
            } else {
                if (img.get(0).complete) {
                    this.updateHeight();
                    this.scrollToPreview();
                } else {
                    img.load(function () {
                        this.updateHeight();
                        this.scrollToPreview();
                    }.bind(this));
                }
            }
        },
```

And about the wrong image being displayed at the image-preview, we probably would have to listen to the data provider and update the image-preview component, that will prevent this issue from happening with bookmarks and other similar scenarios.

Let me know if you have any questions or want to discuss the approaches!

**Comment by sivaschenko** [comment]
Hi @mmezhensky , would you like to continue progress on this pull request? Or do you need any help?

**Comment by mmezhensky** [comment]
Hello @gabrieldagama @sivaschenko 
Sorry, but I didn't have time for this earlier.
Тow I'm back to that, and I created a new commit where I changed fix for the console error and fixed wrong image being displayed at the image-preview.

**Comment by mmezhensky** [comment]
@magento run all tests

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js [hunk 1]
```diff
@@ -48,10 +48,33 @@ define([
                 }
             ],
             listens: {
-                '${ $.sortByComponentName }:applied': 'hide'
+                '${ $.sortByComponentName }:applied': 'hide',
+                'adobe_stock_images_listing.adobe_stock_images_listing.listing_top.bookmarks:activeIndex': 'onActiveIndexChange'
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js:52]: Please add the bookmarksProvider to the defaults to be able to configure it for adobe stock listing and standalone adobe stock listing separately:
 - `AdobeStockImageAdminUi/view/adminhtml/ui_component/standalone_adobe_stock_images_listing.xml`
 - `AdobeStockImageAdminUi/view/adminhtml/ui_component/adobe_stock_images_listing.xml`

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @mmezhensky ! I have added a comment on the implementation, however, this PR does not fix the issue for me. Can you please recheck it

**Comment by sivaschenko** [comment]
Closed due to inactivity

**Pull request closed without merging**